Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -60,183 +70,427 @@ public partial class GestureManager : Singleton<GestureManager>
+            // Start looking for manipulation gestures.
             manipulationRecognizer.StartCapturingGestures();
         }
 
+        #region Interaction Management
+
+        /// <summary>
+        /// Raised when we detect an interaction source.
+        /// </summary>
+        /// <param name="state"></param>
+        private void InteractionManager_SourceDetected(InteractionSourceState state)
+        {
+            trackedInteractionSource.Add(state.source.id);
+        }
+
+        /// <summary>
+        /// Raised when the interaction source is pressed.
+        /// </summary>
+        /// <param name="state">The current state of the Interaction source.</param>
         private void InteractionManager_SourcePressed(InteractionSourceState state)
         {
-            if (!HandPressed)
+            // Make sure we're using a tracked interaction source.
+            if (trackedInteractionSource.Contains(state.source.id))
             {
-                currentHandState = state;
-            }
+                // Add it to the list of pressed states.
+                if (!pressedInteractionSource.Contains(state.source.id))
+                {
+                    pressedInteractionSource.Add(state.source.id);
+                }
 
-            pressedHands.Add(state.source.id);
+                // If we're not currently processing a manipulation then start.
+                if (!ManipulationInProgress)
+                {
+                    // Cache our current source state for use later.
+                    currentInteractionSourceState = state;
+
+                    // Gesture Support for Controllers: (i.e. Clicker, Xbox Controller, etc.)
+                    // Don't start another manipulation gesture if one is already underway.
+                    if (state.source.kind == InteractionSourceKind.Controller)
+                    {
+                        OnManipulation(inProgress: true, offset: ManipulationPosition);
+
+                        if (OnManipulationStarted != null)
+                        {
+                            OnManipulationStarted(state.source.kind);
+                        }
+                    }
+                }
+            }
         }
 
+        /// <summary>
+        /// Raised when the interaction source is updated.
+        /// </summary>
+        /// <param name="state">The current state of the Interaction source.</param>
         private void InteractionManager_SourceUpdated(InteractionSourceState state)
         {
-            if (HandPressed && state.source.id == currentHandState.source.id)
+            // if we currently in a manipulation, update our data.
+            // Check the current interaction source matches our cached value.
+            if (ManipulationInProgress && state.source.id == currentInteractionSourceState.source.id)
             {
-                currentHandState = state;
+                // Gesture Support for Controllers: (i.e. Clicker, Xbox Controller, etc.)
```

Comment:
This was removed to prevent other interaction sources from interfering with the current manipulation.